software-development/SDLC.md

```diff
@@ -0,0 +1,101 @@
+
+Questions typically answered in this phase are. 
+
+-	What tool or technology is most suited for the job:<br/>
+    o	Does the system need to be high available<br/>
+    o	Will there be a high load on the system (a lot of concurrent users)?<br/>
+    o	What data storage do we need to select<br/>
+-	What tools do we have available in the company<br/>
+    o	Some tools (like databases) come with high licenses costs<br/>
+    o	Not all teams can decide themselves what they want to use<br/>
+    o	The design needs to make sure the software can be supported and operated within the organization’s constraints<br/>
+-	How do the different components communicate<br/>
+    o	In a complex system there are multiple applications communicating with each other to complete a business process<br/>
+    o	What data format will be used<br/>
+
+### Software architecture
+
+Because of its importance we spend a little bit more time on the software architecture as it is an essential part of software development. 
+As described on [this website on architecture](https://martinfowler.com/architecture/), Architecture is described as the way the highest level components are wired together. The site also explains the reason why architecture matters:
+
+*Architecture is a tricky subject for the customers and users of software products - as it isn't something they immediately perceive. But a poor architecture is a major contributor to the growth of cruft - elements of the software that impede the ability of developers to understand the software. Software that contains a lot of cruft is much harder to modify, leading to features that arrive more slowly and with more defects.*
+
+It's important to understand that making fundamental changes to the software architecture can be costly; changing from a MongoDB database to Postgress can be complex and time consuming. Therefore it's important to understand that: <br/>
+*Software architecture is about making fundamental structural choices that are costly to change once implemented.*
+
+
+
+**Output of this phase:** a software design <br/>
+**Persons involved:** Engineers, either a (senior) software engineer of the team or an architect<br/>
+
+## Software Development
+**Background**: In this phase the actual coding starts and the system is build. One or multiple engineer start to build the entire system writing code in a chosen programming language. The requirements and the high level architecture design usually serve as input to this phase. As there are usually more than one engineer involved, and in the future other engineers might need to work on the code you created it’s common that the engineers follow the coding guidelines of the company they are working for. 
```

Comment:
`system is built` instead of `system is build`
`One or multiple engineers` instead of `One of multiple engineer`
`work on the code you created. It is common that` instead of `work on the code you created it's common that`